ci: update static check tools

yati1998 opened the pull request:
This PR updates the static check tools to
the latest version.
Further needs to resolve all the errors after
updating the version.

Updates: #1586

Signed-off-by: Yati Padia <ypadia@redhat.com>

yati1998 committed:
ci: update static check tools

This PR updates the static check tools to
the latest version.
Further needs to resolve all the errors after
updating the version.

Updates: #1586

Signed-off-by: Yati Padia <ypadia@redhat.com>

yati1998 commented:
Keeping this PR as draft until all the errors after updating the version are resolved.

nixpanic commented on build.env:
```diff
@@ -16,12 +16,12 @@ BASE_IMAGE=docker.io/ceph/ceph:v15
 
 # standard Golang options
-GOLANG_VERSION=1.15.5
+GOLANG_VERSION=1.16.2
```
this is not a static check tool. If an update of Go is required, explain the reason in the commit message.

yati1998 replied:
okay.

nixpanic commented on build.env:
```diff
@@ -33,7 +33,7 @@ SNAPSHOT_VERSION=v4.0.0
 
 # helm chart generation, testing and publishing
-HELM_VERSION=v3.1.2
+HELM_VERSION=v3.5.4
```
This is also not (only) a static check tool.

nixpanic reviewed (changes requested):
Please split the commit in different tools. That way, addressing the issues identified by a single tool can earlier get merged and  used for other PRs (preventing regressions). Feel free to send them in different PRs too.

nixpanic commented:
@mergifyio rebase

yati1998 committed:
cleanup: gocyclo being unused for linter

This commit addresses the following issue:
'nolint:gocyclo // complexity needs to be reduced.'
is unused for linter "gocyclo" (nolintlint)

Updates:#2025

Signed-off-by: Yati Padia <ypadia@redhat.com>

yati1998 committed:
cleanup: Modifies Wrapcheck linter

Wrapcheck is a  simple Go linter to check that errors
from external packages are wrapped during return to
help identify the error source during debugging.
This commit addresses the wrapcheck error

Updates:#2025

Signed-off-by: Yati Padia <ypadia@redhat.com>

yati1998 committed:
cleanup: address wasted assign issues

At places variable is reassigned without
being used.

Signed-off-by: Yati Padia <ypadia@redhat.com>

Madhu-1 commented on e2e/cephfs_helper.go:
```diff
@@ -179,11 +179,11 @@ func getSnapName(snapNamespace, snapName string) (string, error) {
 	snap, err := sclient.SnapshotV1beta1().VolumeSnapshots(snapNamespace).Get(context.TODO(), snapName, metav1.GetOptions{})
 	if err != nil {
-		return "", err
+		return "", fmt.Errorf("failed to get volume snapshot with error %w", err)
```
volumesnapshot single word

yati1998 replied:
okay.

Madhu-1 commented on e2e/cephfs_helper.go:
```diff
@@ -179,11 +179,11 @@ func getSnapName(snapNamespace, snapName string) (string, error) {
 	sc, err := sclient.SnapshotV1beta1().VolumeSnapshotContents().Get(context.TODO(), *snap.Status.BoundVolumeSnapshotContentName, metav1.GetOptions{})
 	if err != nil {
-		return "", err
+		return "", fmt.Errorf("failed to get volume snapshot content with error %w", err)
```
volumesnapshotcontent single word

Madhu-1 commented on e2e/configmap.go:
```diff
@@ -68,14 +68,14 @@ func createConfigMap(pluginPath string, c kubernetes.Interface, f *framework.Fra
 		_, updateErr := c.CoreV1().ConfigMaps(cephCSINamespace).Update(context.TODO(), &cm, metav1.UpdateOptions{})
 		if updateErr != nil {
-			return updateErr
+			return fmt.Errorf("failed to update %w", updateErr)
```
failed to update configmap?

yati1998 replied:
Yes.  Earlier there was no message so I just added. Will update elsewhere as well.

Madhu-1 commented on e2e/node.go:
```diff
@@ -13,7 +14,7 @@ func createNodeLabel(f *framework.Framework, labelKey, labelValue string) error
 	nodes, err := f.ClientSet.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
 	if err != nil {
-		return err
+		return fmt.Errorf("%w", err)
```
failed to get node list?

Madhu-1 commented on e2e/node.go:
```diff
@@ -24,7 +25,7 @@ func createNodeLabel(f *framework.Framework, labelKey, labelValue string) error
 	nodes, err := c.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
 	if err != nil {
-		return err
+		return fmt.Errorf("%w", err)
```
failed to get node list?

Madhu-1 commented on e2e/node.go:
```diff
@@ -35,7 +36,7 @@ func deleteNodeLabel(c kubernetes.Interface, labelKey string) error {
 	nodes, err := c.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
 	if err != nil {
-		return err
+		return fmt.Errorf("%w", err)
```
failed to get node list

Madhu-1 commented on internal/journal/voljournal.go:
```diff
@@ -470,7 +470,7 @@ func reserveOMapName(ctx context.Context, monitors string, cr *util.Credentials,
 			}
 
-			return "", err
+			return "", fmt.Errorf("failed to create object %w", err)
```
Am thinking ca we disable Wrapcheck feels like this might cause an issue for error return and comparison, Lets hear it from others.

yati1998 replied:
Okay.

nixpanic replied:
Wrapping has my preference. We need to compare returned errors with `errors.Is()` if there is special handling of the errors needed. If comparing errors without `errors.Is()` is done, it needs to be corrected.

yati1998 committed:
cleanup: addresses paralleltest linter

The Go linter paralleltest checks that the t.Parallel
gets called for the test method and for the range of
test cases within the test.

Updates: #2025

Signed-off-by: Yati Padia <ypadia@redhat.com>
